lib/Sema/CSDiagnostics.cpp

```diff
@@ -8199,6 +8199,47 @@ bool UnableToInferClosureReturnType::diagnoseAsError() {
   return true;
 }
 
+bool UnableToInferGenericPackElementType::diagnoseAsError() {
+  auto *locator = getLocator();
+  auto path = locator->getPath();
+
+  const auto applyArgToParamElt =
+      (path.end() - 2)->getAs<LocatorPathElt::ApplyArgToParam>();
+  const auto packElementElt =
+      (path.end() - 1)->getAs<LocatorPathElt::PackElement>();
+  if (!applyArgToParamElt || !packElementElt) {
+    return false;
```

Comment:
We shouldn't produce a fix in this case because this would just result in a fallback diagnostic this looks like more of an assert to me

Reply:
Just so we are on the same page here - you don't have to do it in this PR. This would have to be detected during fixing of the conversion mismatch that's when we have both Set and Array types, we can have a special case in repairFailures for collections

Reply:
Okay, I'll try to look into this. Just one more question – is this supposed to be a generic type mismatch or the specific case for collections? 

This could also invoke the same diagnostics with the same path:

```swift

struct S<T> {
    init() {}
}

struct S2<T> {
    init() {}
}

func patternInstantiationTupleTest1<each T>() -> (repeat S<each T>) {}

func patternInstantiationConcreteInvalid() {
  let _: S2<Int> = patternInstantiationTupleTest1()
  let _: (S<Int>, S2<String>) = patternInstantiationTupleTest1()
}
```

Reply:
Hm, maybe it's okay to forward generic arguments upon mismatches in general if the number of expected arguments on the other side matches, so if both types are BoundGenericTypes with the same number of arguments